agent/command_test.go

```diff
@@ -168,3 +195,56 @@ func TestEndTaskSyncCommands(t *testing.T) {
 		})
 	}
 }
+
+func (s *CommandSuite) TestVarsAreUnsetAfterRunning() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var err error
+	config := &internal.TaskConfig{
+		Expansions: &util.Expansions{"globalVar": "expansionVar"},
+		BuildVariant: &model.BuildVariant{
+			Name: "some_build_variant",
+		},
+		Task: &task.Task{
+			Id:           "task_id",
+			DisplayName:  "some task",
+			BuildVariant: "some_build_variant",
+			Version:      "v1",
+		},
+		Timeout: &internal.Timeout{},
+	}
+	s.tc.taskConfig = config
+	p := &model.Project{
+		Functions: map[string]*model.YAMLCommandSet{
+			"yes": {
+				MultiCommand: []model.PluginCommandConf{
+					{
+						Vars:    map[string]string{"globalVar": "functionVar"},
+						Command: "shell.exec",
+						Params: map[string]interface{}{
+							"script": "echo in the function",
+						},
+					},
+				},
+			},
+		},
+	}
+	s.tc.logger, err = s.mockCommunicator.GetLoggerProducer(ctx, s.tc.task, nil)
+	s.tc.project = p
+	s.tc.taskConfig.Project = p
+
+	func1 := model.PluginCommandConf{
+		Function:    "yes",
+		DisplayName: "function",
+		Vars:        map[string]string{"globalVar": "functionVar"},
+	}
+
+	cmds := []model.PluginCommandConf{func1}
+	err = s.a.runCommandsInBlock(ctx, s.tc, cmds, runCommandsOptions{}, "")
+	s.NoError(err)
+
+	globalVarValue := s.tc.taskConfig.Expansions.Get("globalVar")
+	s.Equal("expansionVar", globalVarValue, "globalVar should be set back to what it was before the function ran")
+	s.NoError(s.tc.logger.Close())
```

Comment:
It's not necessary to `s.tc.logger.Close()` unless you want to read data out of it (this is not true of loggers in production code, but is true in tests that use the mock communicator's logger because the mock logger is just writing to an in-memory buffer).